tests/TestTCP.hs

```diff
@@ -727,7 +726,7 @@ testReconnect = do
     -- But a send will fail because the server has closed the connection again
     WaitSocketFree wait <- takeMVar socketClosed
     wait
-    threadDelay 100000
+    threadDelay 500000
```

Comment:
Oddly enough, if there's no delay, the send succeeds. I'm going to look into this test case and make a separate pull request to fix it. In the mean-time, #60 should be reviewed and merged.